### ISSUE TITLE
writing voxels sizes

### ISSUE THREAD

**Issue by shahpnmlab** [ccpem/mrcfile#26]
Hi Colin,
I am trying to write a volume to disk using mrcfile and setting the header information in the new volume from an existing one by setting directly to the header attribute. This does not appear to be supported but there is no convenient way to serialise/deserialise the rec array that I can see. Would consider adding support for this use case? Alternatively, could you recommend a way to achieve this?

**Comment by colinpalmer** [comment]
Hi Pranav,

No, for better or worse, it was a deliberate decision not to allow the header to be directly copied. The reason was that if you're changing the data, it could easily invalidate some parts of the header so it's better that the user chooses what parts they want to keep or update.

On the other hand, this use case has come up a few times now so maybe there should be explicit support for it. If I get some time I'll take a look at adding it.

For now, the easiest workaround is to just copy the file first, then edit the copy by writing in the new data, which will leave the original header as it was except for the data size fields that are automatically updated from the new data array.

**Comment by shahpnmlab** [comment]
Hi Colin,
Thanks for the info. 
Regarding the title - the thought process started as an observation that I made whilst trying to write the voxel size where I was setting the voxel size prior to setting the data. This resulted in the original voxel size being overwritten. After discussing with a colleague, I learnt that one had to set the voxel size _AFTER_ setting the data array. I found this to be a little unintuitive.
I originally intended to submit this as a bug report, but realised that this is just the normal behaviour of the package and instead ended up asking you about setting the header information directly as it would have saved me from all the trial and error. :)
